File: zeus-api/src/app/modules/project/dtos/verified-code-module.dto.ts
import { ApiProperty } from "@nestjs/swagger";
import { ErrorDto } from "../../visualizer/dtos/error.dto";
import { CodeModuleCounterexampleDto } from "../../visualizer/dtos/code-module-counterexample.dto";

export class VerifiedCodeModuleDto {
  @ApiProperty({type: [CodeModuleCounterexampleDto]})
  counterexamples: CodeModuleCounterexampleDto[]

  @ApiProperty({type: [ErrorDto]})
  errors: ErrorDto[];
}

File: zeus-compiler/src/main/java/zeus/zeuscompiler/rain/dtos/VerifiedCodeModuleDto.java
package zeus.zeuscompiler.rain.dtos;

import zeus.zeuscompiler.thunder.dtos.CodeModuleCounterexampleDto;
import zeus.zeuscompiler.thunder.dtos.ErrorDto;

import java.util.List;
import java.util.Set;

public record VerifiedCodeModuleDto(Set<CodeModuleCounterexampleDto> counterexamples, List<ErrorDto> errors) {
}
